services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/config/MongoDbConfigProperties.java

```diff
@@ -0,0 +1,256 @@
+        return this;
+    }
+
+    /**
+     * Gets the TCP port that the mongodb is listening on.
+     *
+     * @return The port number.
+     */
+    public int getPort() {
+        return port;
+    }
+
+    /**
+     * Sets the TCP port that the mongodb is listening on.
+     * <p>
+     * The default port value is {@link #DEFAULT_PORT}.
+     *
+     * @param port The port number.
+     * @return A reference to this for fluent use.
+     * @throws IllegalArgumentException if port &lt; 1000 or port &gt; 65535.
+     */
+    public MongoDbConfigProperties setPort(final int port) {
+        if (PortConfigurationHelper.isValidPort(port)) {
+            this.port = port;
+        } else {
+            throw new IllegalArgumentException("invalid port number");
+        }
+        return this;
+    }
+
+    /**
+     * Gets the database name.
+     *
+     * @return The database name.
+     */
+    public String getDbName() {
+        return dbName;
+    }
+
+    /**
+     * Sets the database name.
+     *
+     * @param dbName The database name
+     * @return A reference to this for fluent use.
+     * @throws NullPointerException if dbName is {@code null}.
+     */
+    public MongoDbConfigProperties setDbName(final String dbName) {
+        this.dbName = Objects.requireNonNull(dbName);
+        return this;
+    }
+
+    /**
+     * Gets the user name used for authentication.
+     *
+     * @return The user name.
+     */
+    public String getUsername() {
+        return username;
+    }
+
+    /**
+     * Sets the user name used for authentication.
+     *
+     * @param username The user name.
+     * @return A reference to this for fluent use.
+     * @throws NullPointerException if the username is {@code null}.
+     */
+    public MongoDbConfigProperties setUsername(final String username) {
+        this.username = Objects.requireNonNull(username);
+        return this;
+    }
+
+    /**
+     * Gets the password used for authentication.
+     *
+     * @return The password.
+     */
+    public String getPassword() {
+        return password;
+    }
+
+    /**
+     * Sets the password used for authentication.
+     *
+     * @param password the password
+     * @return A reference to this for fluent use.
+     * @throws NullPointerException if password is {@code null}.
+     */
+    public MongoDbConfigProperties setPassword(final String password) {
+        this.password = Objects.requireNonNull(password);
+        return this;
+    }
+
+    /**
+     * Gets the connection string for the mongodb client.
+     *
+     * @return The connection string.
+     */
+    public String getConnectionString() {
+        return connectionString;
+    }
+
+    /**
+     * Sets the connection string for the mongodb client. If set, the connection string
+     * overrides the other connection settings. Format:
+     * mongodb://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
```

Comment:
IMHO a link to the resource where the connection string format is defined would be helpful ...

Reply:
Done.